prog/vm/github_runner.rb

```diff
@@ -223,6 +223,33 @@ def setup_info
       UBICLOUD_CACHE_URL=#{Config.base_url}/runtime/github/" | sudo tee -a /etc/environment
     COMMAND
 
+    if vm.vm_host.id == Vm[Config.docker_mirror_server_vm_id]&.vm_host&.id
+      mirror_server_vm = Vm[Config.docker_mirror_server_vm_id]
+      mirror_server_address = mirror_server_vm.load_balancer.hostname
+      mirror_server_port = 5000
+      command += <<~COMMAND
+        # Configure Docker daemon with registry mirror
+        if [ -f /etc/docker/daemon.json ] && [ -s /etc/docker/daemon.json ]; then
+          jq '. + {"registry-mirrors": ["https://#{mirror_server_address}:#{mirror_server_port}"]}' /etc/docker/daemon.json | sudo tee /etc/docker/daemon.json.tmp
+          sudo mv /etc/docker/daemon.json.tmp /etc/docker/daemon.json
+        else
+          echo '{"registry-mirrors": ["https://#{mirror_server_address}:#{mirror_server_port}"]}' | sudo tee /etc/docker/daemon.json
+        fi
+
+        # Configure BuildKit to use the mirror
+        sudo mkdir -p /etc/buildkit
+        echo '[registry."docker.io"]
+          mirrors = ["#{mirror_server_address}:#{mirror_server_port}"]
+
+          [registry."#{mirror_server_address}:#{mirror_server_port}"]
+          http = false
+          insecure = false' | sudo tee /etc/buildkit/buildkitd.toml
```

Comment:
Should we check if this file exists? Since it's toml, we might want to append to it instead of overwriting it if it exists